integration/mochaTests/README.md

```diff
@@ -0,0 +1,131 @@
+```
+
+### Running Browser Compatibility Tests on DeviceFarm
+To run the tests on DeviceFarm, you will need an AWS account. Once you have an AWS account, you will need to setup two environment variables that will allow the AWS SDK to authenticate to your AWS account.
+```bash
+export AWS_ACCESS_KEY_ID=<YOUR_ACCESS_KEY_ID>
+export AWS_SECRET_ACCESS_KEY=<YOUR_SECRET_ACCESS_KEY>
+```
+
+You will need to create a DeviceFarm project in the AWS account that you are planning to use. There are two types of devicefarm projects that can be created: `Mobile Device Testing` and `Desktop Browser Testing`. For this section, we will focus mainly on desktop browser testing.
+After the project is created you will need to setup a `PROJECT_ARN` as an environment variable. Project arn is used by the device farm API to identify the project to create test sessions inside.
+```bash
+export PROJECT_ARN=<YOUR_PROJECT_ARN>
+```
+
+Following command can be used to run a browser compatibility test with default settings on DeviceFarm:
+```bash
+npm run test -- --test-name AudioTest --host devicefarm --test-type browser-compatibility
+```
+
+Just like with SauceLabs, DeviceFarm can run browser compatibility tests with a custom config:
+```bash
+npm run test -- --test-name AudioTest --host devicefarm --test-type browser-compatibility --config browserCompatibilityTest/desktop/sample_test.config.json
+```
+
+## Logging
+
+In the integration tests, logging is implemented in a unique way. This section will go over the two distinct ways of logging:
+1. Use `logger.log` for inline logging
+2. Use `logger.pushLogs` for buffered logging
+
+Developers can use `logger.log` as a normal console log with some enhanced functionality. The main difference between `logger.log` and `console.log` is that logger log will take `LogLevel` as a parameter. Different log levels print to the console in different colors and have a minor difference in the logs which makes it easier to read the logs. There are four different types of log levels.
+```ts
+const LogLevel = {
+  INFO: 'INFO',
+  WARN: 'WARN',
+  ERROR: 'ERROR',
+  SUCCESS: 'SUCCESS',
+};
+```
+
+Integration tests use `mocha` as their test framework. Mocha prints the test results and some limited logs to the console automatically. Any console logs that are inside of `it` will print before the mocha logs.
+This makes debugging hard as the logs seem out of sync and random. To get around this issue buffered logging was added. Inside of `it` blocks, `logger.pushLogs` can be used to add logs to the buffer and a call to `logger.printLogs` inside of the `afterEach` hook will print the logs in the desired order.
+```ts
+afterEach(async function () {
+  this.logger.printLogs();
+});
+``` 
+
+Mocha provides hooks like `before`, `after`, `beforeEach`, `afterEach` to perform actions at specific time of the testing cycle. Any logs printed inside the hooks will print in line so inline logging can be used and there is no need for buffered logging.
```

Comment:
- `at specific time` => `at a specific time`
- `in line so` => `in line, so` (add a comma)